cmd/ztest/ztest.c

```diff
@@ -3529,13 +3528,9 @@ ztest_device_removal(ztest_ds_t *zd, uint64_t id)
 	uint64_t guid;
 	int error;
 
+	mutex_enter(&ztest_removal_lock);
```

Comment:
the old code was effectively a tryenter, do we want to use that here?

Reply:
This change was also from @behlendorf. I think the idea here was to make sure that this code path actually gets to run. As he mentioned, we might want to change this since a lot of these code paths are actually valid to race against each other in the actual kernel module.

Reply:
We may, in fact as I mentioned in my comment,  think we should hold off entirely on the boolean -> lock conversion patch moment.  It's intention was to prevent racing between fault injection and removal, but we don't actually have clear evidence yet that it's needed.